packages/apidom-ls/src/services/validation/linter-functions.ts

```diff
@@ -1068,4 +1069,52 @@ export const standardLinterfunctions: FunctionItem[] = [
       return true;
     },
   },
+  {
+    functionName: 'apilintOpenAPIParameterFieldIsDefinedWithinPathTemplate',
+    function: (element: Element) => {
+      if (element.element === 'parameter') {
+        const parameterName = toValue((element as ObjectElement).get('name'));
+        const parameterLocation = toValue((element as ObjectElement).get('in'));
```

Comment:
We can bail out right here, if it's not a path parameter:

```js
        if (parameterLocation !== 'path') {
          return true;
        }
```

Reply:
> Just the path check isn't enough as Parameter objects defined in components can also have in with value path and it passes this check.

We're not interested in those Parameter Objects as they are not used in the process of path templating. And yes it should pass the check (return true).

>  I'm not sure it's valid OpenAPI definition, but if it is, it has to be accounted for.

It's valid OpenAPI definition, and it's accounted for with checks mentioned in previous comment.

Reply:
> It's valid OpenAPI definition, and it's accounted for with checks mentioned in https://github.com/swagger-api/apidom/pull/3571#discussion_r1440635491.

Yeah I was just trying to say that the checko below:
```javascript
 if (parameterLocation !== 'path') {
     return true;
  }
``` 
Isn't enough on it's own. 

> We can avoid step 1. and all the following steps by immediately looking into Parameter.in values and checking if it's path.

The above tip from the initial comment confused me a bit and I thought for a while that just this check would be enough, but as discussed, it isn't.

Reply:
No worries, this is one of the most complex linting function. We have to be synced with how we want to approach the problem and why.

If something is ambiguous in https://github.com/swagger-api/apidom/pull/3571#issuecomment-1873771768, we can have a zoom call and sync.